subworkflows/local/bam_variant_calling_somatic_controlfreec/main.nf

```diff
@@ -25,18 +25,44 @@ workflow BAM_VARIANT_CALLING_SOMATIC_CONTROLFREEC {
 
     ch_versions = Channel.empty()
 
+    assess_significance_input = Channel.empty()
+
     FREEC_SOMATIC(controlfreec_input, fasta, fasta_fai, [], dbsnp, dbsnp_tbi, chr_files, mappability, intervals_bed, [])
 
-    ASSESS_SIGNIFICANCE(FREEC_SOMATIC.out.CNV.join(FREEC_SOMATIC.out.ratio, failOnDuplicate: true, failOnMismatch: true))
+    //Filter the files that come out of freec somatic as ASSESS_SIGNIFICANCE only takes one cnv and one ratio file
+    //Creates empty channel if file is missing
+    cnv_files = FREEC_SOMATIC.out.CNV
+    .multiMap{ meta, cnv ->
```

Comment:
I suppose one could also get away with using just a `.map()` here instead of a `.multiMap()`, right?  Perhaps like this?

```
    cnv_files_tumor = FREEC_SOMATIC.out.CNV
    .map{ meta, cnv ->
        def meta_clone_tumor = meta.clone()
        def tumor_file = cnv instanceof List ? cnv.find { it.toString().endsWith("gz_CNVs") } : cnv //only find if its a list, else it returns only the filename without the path

        def foo = tumor_file == null ? [] : [meta_clone_tumor,tumor_file] //only fill channel if file was found, else leave it empty
        foo
    }
```
Anyways, I guess we don't pay extra for using multiMaps? 😆

Reply:
Hi @asp8200 You are absolutely right, having the assess_significance_input empty channel definition and using multiMap are not required. I placed them there because, originally, I had made the modification so that the "FREEC_SOMATIC.out.CNV" channel would be splitted into tumor_files and normal_files, and it would run the assess_significance_input module on both. But recently I was told that we only needed to run it on the tumor_files. I left it there in case we ever wanted to run it on the normal files as well, so it would be easier to implement also, as you say, it doesn't affect the execution. But please let me know if you would prefer that I implement these changes you mentioned!

Reply:
You're welcome to leave the multiMap. I would get rid of `assess_significance_input` but it is really not important. Good work, @nschcolnicov 👏